### PR TITLE
Updated C++ quickstart guide

### LINKED ISSUE #237: Refresh Quickstart content for website.
- [x] Volunteers for every language. 
- [x] Review content, provide PRs if necessary.

### PR DESCRIPTION
**Pull request by dgquintas**
We also need to:

1. Update grpc/INSTALL.md (mention instructions for windows, for instance).
2. Update https://github.com/grpc/grpc/tree/master/examples/cpp/README.md, probably to simply point to this file. Or make it a verbatim copy.

Fixes #237 for C++

### CODE REVIEW

File: docs/quickstart/cpp.md [hunk 1]
```diff
@@ -1,7 +1,280 @@
...
+[here](https://github.com/grpc/grpc/blob/master/INSTALL.md).
+
+#### Install Protocol Buffers v3
+
+While not mandatory, gRPC usually leverages Protocol Buffers v3 for service
+definitions and data serialization. If you don't already have it installed on
+your system, you can install the version cloned alongside gRPC:
+
+```sh
+$ git clone -b $(curl -L http://grpc.io/release) https://github.com/grpc/grpc
```
Comment (hsaliak) [inline docs/quickstart/cpp.md]: Can we change this to curl -L https://grpc.github.io/release

File: docs/quickstart/cpp.md [hunk 2]
```diff
@@ -1,7 +1,280 @@
...
+$ cd grpc/third_party/protobuf
+$ sudo make install
```
Comment (hsaliak) [inline docs/quickstart/cpp.md]: I see that we are not making any reference to the compile dependencies here. Is that deliberate? 
Should we reference "For detailed installation instructions, refer to their README.md"?

File: docs/quickstart/cpp.md [hunk 3]
```diff
@@ -1,7 +1,280 @@
...
+## Download the example
+
+You'll need a local copy of the example code to work through this quickstart.
+You may already have this from the installation step. Otherwise, download the
+example code from our Github repository (the following command clones the entire
+repository, but you just need the examples for this quickstart and other
+tutorials):
+
+```sh
+$ git clone -b $(curl -L http://grpc.io/release) https://github.com/grpc/grpc
```
Comment (hsaliak) [inline docs/quickstart/cpp.md]: Could we also change this to https://grpc.github.io/release

File: docs/quickstart/cpp.md [hunk 4]
```diff
@@ -1,7 +1,280 @@
...
+
+#### Install gRPC
+
+To install gRPC on your system, follow the instructions to build from source
+[here](https://github.com/grpc/grpc/blob/master/INSTALL.md).
+
+#### Install Protocol Buffers v3
+
+While not mandatory, gRPC usually leverages Protocol Buffers v3 for service
+definitions and data serialization. If you don't already have it installed on
```
Comment (dgquintas) [inline docs/quickstart/cpp.md]: Well, this is inside the "Prerequisites" section.

File: docs/quickstart/cpp.md (hunk 1 above)
Comment (dgquintas) [inline docs/quickstart/cpp.md]: Yes. Can I do that globally on a separate PR?

File: docs/quickstart/cpp.md (hunk 2 above)
Comment (dgquintas) [inline docs/quickstart/cpp.md]: which dependencies? for the compilation/installation I'm mostly pointing the reader to grpc/INSTALL.md

File: docs/quickstart/cpp.md (hunk 3 above)
Comment (dgquintas) [inline docs/quickstart/cpp.md]: ditto

File: docs/quickstart/cpp.md (hunk 1 above)
Comment (hsaliak) [inline docs/quickstart/cpp.md]: Yup!

File: docs/quickstart/cpp.md (hunk 4 above)
Comment (hsaliak) [inline docs/quickstart/cpp.md]: Noted.

File: docs/quickstart/cpp.md (hunk 2 above)
Comment (dgquintas) [inline docs/quickstart/cpp.md]: done. ptal.

File: docs/quickstart/cpp.md (hunk 1 above)
Comment (dgquintas) [inline docs/quickstart/cpp.md]: alright, i'll do it once this PR gets merged.

**Comment by hsaliak** [comment]
LGTM
